pkg/autoscaler/statserver/server_test.go

```diff
@@ -189,6 +190,45 @@ func TestServerDoesNotLeakGoroutines(t *testing.T) {
 	}
 }
 
+func TestServerWithOwnershipStatsReceived(t *testing.T) {
+	statsCh := make(chan metrics.StatMessage)
+	server := newTestServerWithOwnership(statsCh, &testOwnership{})
+	defer server.Shutdown(0)
+
+	go server.listenAndServe()
+
+	// Verify the server behavior normally when the host is not a bucket host
+	statSink := dialOK(t, server.listenAddr())
+	assertReceivedProto(t, both, statSink, statsCh)
+	closeSink(t, statSink)
+}
+
+func TestServerWithOwnershipConnectionClosed(t *testing.T) {
+	// Override the function to mock a bucket host.
+	isBucketHost = func(host string) bool { return true }
```

Comment:
Can we instead mock the request to have `autoscaler-bucket` in it to avoid the shenanigans with the globals?

Reply:
I tried that at first but couldn't find a way to do it. I guess we have to hack the pkg/websocket lib. Let me know if there's a way to mock the request host.